src/entity/posts/utils.ts

```diff
@@ -285,6 +288,96 @@ interface CreateExternalLinkArgs {
   };
 }
 
+interface PreparePostContext {
+  con: DataSource | EntityManager;
+  userId?: string;
+  req?: Pick<FastifyRequest, 'ip'>;
+}
+
+/**
+ * Prepares a post for insertion by checking all vordr conditions
+ * and setting appropriate flags if the post should be shadow banned.
+ * This should be called before saving any new post.
+ */
+export const preparePostForInsert = async <T extends DeepPartial<Post>>(
+  post: T,
+  context: PreparePostContext,
+): Promise<T> => {
+  const postContent = (post as DeepPartial<FreeformPost>).content; // Some post types have content
+  const shouldVordr = await checkWithVordr(
+    {
+      id: post.id || 'new-post',
+      type: VordrFilterType.Post,
+      title: post.title || undefined,
+      content: postContent,
+    },
+    context,
+  );
+
+  if (shouldVordr) {
+    logger.info({ postId: post.id }, 'Post will be vordr on insert');
```

Comment:
again not sure its useful to have logs, you can always see in the DB if post is vordr?